themes/default/content/blog/log4shell-response/index.md

```diff
@@ -0,0 +1,121 @@
+like transforming that input into a string. That process is known as
+sanitization, and it's a very common security step. You should sanitize input
+with your Pulumi program as well if you're using something like the automation
+API that takes input from other systems or a command line interface.
+
+### Isolate your concerns
+
+If you are not in control of that sanitization, you should attempt to separate
+your concerns by isolating your logging system to a write-only archive. This
+means that the logging system should not be able to write out to any system, and
+no system should be able to edit any logs that are sent to that system. How does
+that help? One of the most common concerns with the Log4Shell is the ability of
+the JNDI to connect to any Lightweight Directory Access Protocol (LDAP) server
+and therefore make calls against any interconnected component of a system. If
+the logging system is isolated so that it cannot trigger any program, process,
+or workflow, it should also not be able to make any call out to the internet
+save perhaps from an approved list of servers (e.g., your PagerDuty endpoint).
+
+### Consider deploying a WAF
+
+One way to ensure that your logging system, or any part of your network really,
+is unable to connect with or be connected to the internet in unexpected ways is
+to add a web application firewall (WAF). Many of the common WAFs out there have
+already deployed rules to block the JNDI Lookup's most common problem spots.
+
+In the registry, we've got a couple of different WAF providers you could add to
+your infrastructure to help block that traffic.
+[Cloudflare](https://www.pulumi.com/registry/packages/cloudflare/) and
+[AWS WAF](https://www.pulumi.com/registry/packages/aws/api-docs/waf/webacl/)
+are just two examples, and the basics of calling each one can be found in those
+docs. Both
+[Cloudflare](https://blog.cloudflare.com/cve-2021-44228-log4j-rce-0-day-mitigation/)
+and [AWS](https://aws.amazon.com/security/security-bulletins/AWS-2021-005/) have
+detailed their mitigation strategies in various posts over the past few days if
+you want to read more.
+
+---
+
+All in all, there is always risk when using third-party tools. The idea is to
+mitigate the risk as much as possible through careful consideration of possible
+attack vectors to evaluate tradeoffs, knowledge of the ring of trust, protection
+through sandboxing or isolating systems that are unrelated, and a plan for
+eventual breaches.
+
+## If you need to remediate
+
+Please note that we are not a logging provider, and we are not the experts on
+the subject, just fellow users. If you are looking for remediation help for your
+systems that directly run log4j, please refer to the [Apache Log4j Security
+Vulnerabilities page](https://logging.apache.org/log4j/2.x/security.html) for
+proper remediation steps and the official [CVE
+record](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2021-44228) for more
```

Comment:
and also CVE-2021-45046 at https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2021-45046

Reply:
And again, thank you!!